### ISSUE TITLE
Add support for oEmbed

### ISSUE THREAD

**Issue by TobiasSN** [forem/forem#6369]
**Is your feature request related to a problem? Please describe.**

Currently, the only way to embed content from other sites into posts is to pray that there's a liquid tag for it, which isn't always the case.

**Describe the solution you'd like**

I propose that Forem gets support for oEmbed. For those who are unaware, oEmbed is a format that allows sites to specify how their content should look when embedded on other sites. You can learn more about it here: https://oembed.com/

I have 2 suggestions for how people might use it:
1. Just put the URL there and have Forem check if the URL can be embedded using oEmbed. This might not be a good idea, as some posts include a lot of links, which would mean one HTTP request for each, which can get a bit slow.
2. Have a special liquid tag, maybe `{% oembed <url> %}`. This is what I would recommend, as only URLs which are supposed to use oEmbed will try to do so.

**Describe alternatives you've considered**

An alternative could just be to allow iframes. However, this might not work with all sites, yet I still think it's a viable option. [Here's an older issue for it, which seems to have been mostly ignored.](https://github.com/thepracticaldev/dev.to/issues/1445) Note that this issue is not a duplicate of that one, as it's about iframes, while this one is about oEmbed.

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by jitendra1998** [comment]
Hi @rhymes, I am interested in working on this feature.

**Comment by jitendra1998** [comment]
While trying to figure out a solution for issue #1445, read about oembed and created a liquid tag for it using ruby-oembed.

Here is an example of amcharts.com chart in dev.to editor using oembed liquid tag,
![Demo amchart](https://user-images.githubusercontent.com/13114810/79053398-b4c60a80-7c5a-11ea-9519-3d06f621f422.png)

A related issue for the ability to add charts in dev.to [#6362](https://github.com/thepracticaldev/dev.to/issues/6362)

Also researching about other alternatives like embed.ly as well.

**Comment by jitendra1998** [comment]
@rhymes, can I work on it?

**Comment by rhymes** [comment]
hi @jitendra1998, this hasn't been approved yet! :-) Thanks for the inputs in the meantime!

**Comment by lanthias** [comment]
I would love this. [We](https://github.com/datapane/datapane) embed into Medium using Embedly, and oEmbed elsewhere (Notion etc.), and would love to embed into dev.to as well. Hopefully this ticket could also cut down the developer hours required for new embeds, as a lot of folks already support oEmbed/embedly out of the box.

**Comment by cmgorton** [comment]
Hello! Thanks for your contributions here.  I am going to close this issue.  We have a new process moving forward to address feature requests.
TLDR: we have new internal RFC process for feature requests. Any features that are more substantial and require more thought and detail will need to be discussed in [forem.dev](forem.dev).

You can read more about our Internal RFC Process and forem.dev Discussions in [this post](https://forem.dev/foremteam/internal-rfc-process-and-forem-dev-discussions-3gl4).
